tests/Dataset.test.ts

```diff
@@ -54,14 +40,120 @@ describe("Dataset", () => {
+      // No metadata for feature4
+      feature5: { type: "categorical", categories: ["small", "medium", "large"] },
+    },
+  };
+
+  const manifestsToTest: [string, AnyManifestFile][] = [
+    ["Default Manifest", defaultManifest],
+    ["Deprecated Manifest V1", manifestV1],
+  ];
+
+  // Test both normal and deprecated manifests
+  for (const [name, manifest] of manifestsToTest) {
+    describe(name, () => {
+      it("retrieves feature units", async () => {
+        const dataset = new Dataset(defaultPath, undefined, new MockArrayLoader());
+        const mockFetch = makeMockFetchMethod(defaultPath, manifest);
+        await dataset.open(mockFetch);
 
-    expect(dataset.getFeatureUnits("feature1")).to.equal("meters");
-    expect(dataset.getFeatureUnits("feature2")).to.equal("(m)");
-    expect(dataset.getFeatureUnits("feature3")).to.equal("μm/s");
-    expect(dataset.getFeatureUnits("feature4")).to.equal("");
-  });
+        expect(dataset.getFeatureUnits("feature1")).to.equal("meters");
+        expect(dataset.getFeatureUnits("feature2")).to.equal("(m)");
+        expect(dataset.getFeatureUnits("feature3")).to.equal("μm/s");
+        expect(dataset.getFeatureUnits("feature4")).to.equal("");
+        expect(dataset.getFeatureUnits("feature5")).to.equal("");
+      });
+
+      it("retrieves feature types", async () => {
+        const dataset = new Dataset(defaultPath, undefined, new MockArrayLoader());
+        const mockFetch = makeMockFetchMethod(defaultPath, manifest);
+        await dataset.open(mockFetch);
+
+        expect(dataset.getFeatureType("feature1")).to.equal(FeatureType.CONTINUOUS);
+        expect(dataset.getFeatureType("feature2")).to.equal(FeatureType.DISCRETE);
+        expect(dataset.getFeatureType("feature5")).to.equal(FeatureType.CATEGORICAL);
+      });
+
+      it("defaults type to continuous if no type or bad type provided", async () => {
+        const dataset = new Dataset(defaultPath, undefined, new MockArrayLoader());
+        const mockFetch = makeMockFetchMethod(defaultPath, manifest);
+        await dataset.open(mockFetch);
+
+        expect(dataset.getFeatureType("feature3")).to.equal(FeatureType.CONTINUOUS);
+        expect(dataset.getFeatureType("feature4")).to.equal(FeatureType.CONTINUOUS);
+      });
+
+      it("gets feature categories", async () => {
+        const dataset = new Dataset(defaultPath, undefined, new MockArrayLoader());
+        const mockFetch = makeMockFetchMethod(defaultPath, manifest);
+        await dataset.open(mockFetch);
+
+        expect(dataset.isFeatureCategorical("feature1")).to.be.false;
+        expect(dataset.isFeatureCategorical("feature2")).to.be.false;
+        expect(dataset.isFeatureCategorical("feature3")).to.be.false;
+        expect(dataset.isFeatureCategorical("feature4")).to.be.false;
+        expect(dataset.isFeatureCategorical("feature5")).to.be.true;
+        expect(dataset.getFeatureCategories("feature5")).to.deep.equal(["small", "medium", "large"]);
```

Comment:
nit but I missed this function call on the first time reading it. It seems like a public function like this should be its own unit tests, and be checking for the cases you have below, but on the function instead of the whole class.